lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -4300,14 +4300,18 @@ swift::TypeBase *SwiftASTContext::ReconstructType(ConstString mangled_typename,
                    .getPointer();
 
   if (found_type) {
-    found_type =
-        convertSILFunctionTypesToASTFunctionTypes(found_type).getPointer();
-    CacheDemangledType(mangled_typename, found_type);
-    CompilerType result_type = ToCompilerType(found_type);
-    assert(&found_type->getASTContext() == ast_ctx);
+    swift::TypeBase *ast_type =
+        ConvertSILFunctionTypesToASTFunctionTypes(found_type).getPointer();
+    // This transformation is lossy: all SILFunction types are mapped
```

Comment:
Can we just get rid of this mapping? IIRC, the old type reconstruction code didn't handle SILFunctionType at all, always returning `() -> ()`. When I implemented full type reconstruction, there were a few places in lldb that weren't set up to handle a SILFunctionType so I added this hack to get the old behavior back. We should just remove it.